tests/integration/Error/ErrorCollectionTest.php

```diff
@@ -93,4 +93,20 @@ public function testCount_returnsTheNumberOfErrors()
         $this->assertEquals(false, $result->success);
         $this->assertEquals(4, count($result->errors));
     }
+
+    public function testJsonSerialize()
+    {
+        $result = Braintree\Customer::create([
+            'email' => 'invalid',
+            'creditCard' => [
+                'number' => 'invalid',
+                'expirationDate' => 'invalid',
+                'billingAddress' => [
+                    'countryName' => 'invaild'
+                ]
+            ]
+        ]);
+        $this->assertEquals(false, $result->success);
+        $this->assertJsonStringEqualsJsonString(json_encode($result->errors->deepAll()), json_encode($result->errors));
```

Comment:
I am confused how this assertion will work. I am not super familiar with PHP, but ...

shouldn't the left side of the comparison lack the `json_encode` call?

Reply:
I think it's just asserting that the result of encoding the `deepAll()` result is the same as encoding the top level `errors` (which it wasn't before)

Reply:
Yeah, that was my understanding as well, since `deepAll()` returns a flat array, you'd still want to json encode it to check that encoding the top level `errors` (which now implements the `jsonSerialize()` method that also calls `deepAll()` under the hood) returns the same value when encoded.

Reply:
Gotcha! Thank you

Reply:
👌